abacustest/lib_model/model_017_vacancy.py

```diff
@@ -323,10 +335,31 @@ def prepare_vacancy_jobs(
         original_kpt_file = os.path.join(job, input_params.get('kpt_file', 'KPT'))
 
         # Prepare input files for the supercell
+        # cell-relax job for the original structure
         original_stru_jobpath = os.path.join(job, "vacancy_original_stru")
         os.makedirs(original_stru_jobpath, exist_ok=True)
-        copy_pp_orb_kpt_file(job, original_stru_jobpath, pp_orb_files, original_kpt_file)
-        write_inputs(original_stru_jobpath, input_params, original_stru)
+        create_new_abacus_job(job,
+                              original_stru_jobpath,
+                              pp_orb_files,
+                              extra_input=input_params,
+                              stru=original_stru, 
+                              kpt_file=original_kpt_file)
+        # scf job after cell-relax job for the original structure
+        if scf_kspacing is None:
+            if 'kspacing' not in input_params.keys():
+                original_kpt = ReadKpt(os.path.join(original_stru_jobpath, "KPT"))
+                original_kspacing = kpt2kspacing(original_kpt, original_stru.get_cell())
```

Comment:
The return of ReadKpt is a tuple of the K POINTS and model, should put `original_kpt[0][:3]` as the parameter of kpt2kspacing().